### PR TITLE
Adding generated terra-aggregator package

### PR DESCRIPTION
**Pull request by tbiethman**
### Summary
Adding generated terra-aggregator package.

### CODE REVIEW

File: packages/terra-aggregator/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    "url": "https://github.com/cerner/terra-framework/issues"
+  },
+  "homepage": "https://github.com/cerner/terra-framework#readme",
+  "devDependencies": {
+    "terra-props-table": "^1.10.1"
+  },
+  "peerDependencies": {
+    "react": "^15.4.2",
+    "react-dom": "^15.4.2",
+    "terra-base": "^2.0.0"
```
Comment (jeffmerten) [inline packages/terra-aggregator/package.json]: Why is terra-base in both dependency groups?

File: packages/terra-aggregator/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "compile": "npm run compile:clean && npm run compile:build",
+    "compile:clean": "rimraf lib",
+    "compile:build": "babel src --out-dir lib --copy-files",
+    "lint": "npm run lint:js && npm run lint:scss",
+    "lint:js": "eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
+    "lint:scss": "stylelint src/**/*.scss",
+    "props-table": "props-table ./src/Aggregator.jsx --out-dir ./docs/props-table",
+    "test": "npm run test:jest && npm run test:nightwatch",
+    "test:jest": "jest ./tests/jest/* --config ../../jestconfig.json",
+    "test:nightwatch": "nightwatch -c ../../nightwatch.conf.js"
```
Comment (jeffmerten) [inline packages/terra-aggregator/package.json]: Should this be wdio?

File: packages/terra-aggregator/tests/nightwatch/AggregatorTests.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable import/no-extraneous-dependencies */
```
Comment (emilyrohrbough) [inline packages/terra-aggregator/tests/nightwatch/AggregatorTests.jsx]: File not needed.

File: packages/terra-aggregator/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Terra Aggregator
+
+
+[![NPM version](http://img.shields.io/npm/v/terra-aggregator.svg)](https://www.npmjs.org/package/terra-aggregator)
+[![Build Status](https://travis-ci.org/cerner/terra-framework.svg?branch=master)](https://travis-ci.org/cerner/terra-framework)
+
+{insert description}
```
Comment (bjankord) [inline packages/terra-aggregator/README.md]: Description needed

File: packages/terra-aggregator/docs/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
+# Terra Aggregator
+
+{insert description}
```
Comment (bjankord) [inline packages/terra-aggregator/docs/README.md]: Description needed

File: packages/terra-aggregator/package.json [hunk 6]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "terra-aggregator",
+  "private": true,
+  "main": "lib/Aggregator.js",
+  "version": "0.0.0",
+  "description": "{insert description}",
```
Comment (bjankord) [inline packages/terra-aggregator/package.json]: Description needed

**Comment by jeffmerten** [comment]
According to webster, this is the definition of an aggregator:
> someone or something that gathers together materials from a variety of sources

From the conversations we've had, this new component would be responsible for managing focus and selection for child components. Those children would be provided by something else, which actually would be more of an "aggregator" than this new component.

**Comment by emilyrohrbough** [comment]
@jeffmerten Did you have a name suggestion?

**Comment by jeffmerten** [comment]
terra-orchestrator? or terra-focus-manager?

**Comment by mjhenkes** [comment]
I prefer aggregator, it's already in common usage.

**Comment by emilyrohrbough** [comment]
New package structure goes as follows:
```
component
    /docs
    /examples
        /test-examples  # what use to be in the nightwatch dir
        Index.site-page.js
    /src
    /tests
```

**Comment by tbiethman** [comment]
I'm going to go ahead and close this and put out the aggregator PR as a whole. I've taken the comments here into account for the new PR. Many thanks to the folks who took a look.

**Pull request closed without merging**